.circleci/create_circleci_config.py

```diff
@@ -100,14 +106,22 @@ def to_dict(self):
             job["resource_class"] = self.resource_class
         if self.parallelism is not None:
             job["parallelism"] = self.parallelism
+
+        checksum = self.checksum if self.checksum is not None else '{{ checksum "setup.py" }}'
+        save_cache = True
+        if self.checksum is not None:
+            # `setup.py` is not modified and we are not on `main` branch
```

Comment:
`self.checksum` is not None only if:
  - the triggered event is not from `main` branch
  - `setup.py` is not modified